From 2fd0a43bfe9e3e53fe566759d6c2cb63ba557ad3 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 28 Jul 2021 09:49:21 +0100
Subject: [PATCH] char: vc_mem: Delete dead code

There are no error exists once device_create has succeeded, and
therefore no need to call device_destroy from vc_mem_init.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 drivers/char/broadcom/vc_mem.c | 2 --
 1 file changed, 2 deletions(-)

--- a/drivers/char/broadcom/vc_mem.c
+++ b/drivers/char/broadcom/vc_mem.c
@@ -333,8 +333,6 @@ vc_mem_init(void)
 	vc_mem_inited = 1;
 	return 0;
 
-	device_destroy(vc_mem_class, vc_mem_devnum);
-
 out_class_destroy:
 	class_destroy(vc_mem_class);
 	vc_mem_class = NULL;
